### PR TITLE
Formally deprecate the Regular Subst Tactic option.

### PR DESCRIPTION
**Pull request by ppedrot**
It was introduced in 8.5 and turned on by default in 8.6.

See #13498.

- [x] Added / updated test-suite
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

File: doc/changelog/07-vernac-commands-and-options/14336-deprecate-regular-subst-option.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- **Deprecated:**
+  the Regular Subst Tactic option
```
Comment (Zimmi48) [inline doc/changelog/07-vernac-commands-and-options/14336-deprecate-regular-subst-option.rst:2]: More precisely, this is a flag. And until it's removed, you can link to its documentation using `:flag:`.

**Review by Zimmi48 — APPROVED** [review]

File: doc/changelog/07-vernac-commands-and-options/14336-deprecate-regular-subst-option.rst (hunk 1 above)
Comment (ppedrot) [reply doc/changelog/07-vernac-commands-and-options/14336-deprecate-regular-subst-option.rst:2]: Hum, linking it means that when we remove it we will have to fix the changes file. Is that reasonable? I know we've been doing that already but it seems quite ugly to me.

Comment (ppedrot) [reply doc/changelog/07-vernac-commands-and-options/14336-deprecate-regular-subst-option.rst]: (Done anyways.)

**Comment by Zimmi48** [comment]
Will merge tomorrow.

**Comment by Zimmi48** [comment]
@coqbot merge now

**Pull request merged**